src/__tests__/components/buttons/TransactionFormButton.test.tsx

```diff
@@ -228,6 +229,53 @@ describe('TransactionFormButton', () => {
     });
   });
 
+  it('uses latest default values always when update', async () => {
+    jest.spyOn(Transaction, 'findOneOrFail').mockResolvedValue({
+      currency: {
+        mnemonic: 'EUR',
+      },
+      splits: [] as Split[],
+    } as Transaction);
+
+    const { rerender } = render(
+      <TransactionFormButton
+        action="update"
+        defaultValues={{
+          description: 'hello',
+        } as FormValues}
+        account={{
+          guid: '1',
+          name: 'account',
+        } as Account}
+      />,
+    );
+
+    rerender(
+      <TransactionFormButton
+        action="update"
+        defaultValues={{
+          description: 'haha',
+        } as FormValues}
+        account={{
+          guid: '1',
+          name: 'account',
+        } as Account}
+      />,
+    );
+
+    const button = await screen.findByRole('button', { name: /add transaction/i });
+    fireEvent.click(button);
+
+    await waitFor(() => expect(TransactionForm).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        defaultValues: expect.objectContaining({
+          description: 'haha',
```

Comment:
Before this change, the value was `hello` as `React.useState` was setting the state only on the first render and subsequent renders were not updating its value.